include/Hdlcpp.hpp

```diff
@@ -221,14 +222,14 @@ class Hdlcpp {
         return destination.size();
     }
 
-    int decode(Frame &frame, uint8_t &sequenceNumber, const std::vector<uint8_t> &source, uint8_t *destination, uint16_t destinationLength, uint16_t &discardBytes)
+    int decode(Frame &frame, uint8_t &sequenceNumber, const std::span<uint8_t> source, std::span<uint8_t> destination, uint16_t &discardBytes)
```

Comment:
if size_t is used as return type for encode, think we should align it here also?